docs/UsersGuide/source/BuildSRW.rst

```diff
@@ -0,0 +1,489 @@
+
+.. note::
+   The :ref:`container approach <QuickstartC>` is recommended for a smoother first-time build and run experience. Building without a container may allow for more customization. However, the non-container approach requires more in-depth system-based knowledge, especially on Level 3 and 4 systems, so it is less appropriate for beginners. 
+
+To build the SRW App, users will complete the following steps:
+
+   #. :ref:`Install prerequisites <HPCstackInfo>`
+   #. :ref:`Clone the SRW App from GitHub <DownloadSRWApp>`
+   #. :ref:`Check out the external repositories <CheckoutExternals>`
+   #. :ref:`Set up the build environment and build the executables <BuildExecutables>`
+
+
+.. _HPCstackInfo:
+
+Install the HPC-Stack
+========================
+
+.. Attention::
+   Skip the HPC-Stack installation if working on a `Level 1 system <https://github.com/ufs-community/ufs-srweather-app/wiki/Supported-Platforms-and-Compilers>`_ (e.g., Cheyenne, Hera, Orion, NOAA Cloud).
+
+**Definition:** :term:`HPC-Stack` is a repository that provides a unified, shell script-based build system to build the software stack required for `UFS <https://ufscommunity.org/>`_ applications such as the SRW App. 
+
+Background
+----------------
+
+The UFS Weather Model draws on over 50 code libraries to run its applications. These libraries range from libraries developed in-house at NOAA (e.g., NCEPLIBS, FMS) to libraries developed by NOAA's partners (e.g., PIO, ESMF) to truly third party libraries (e.g., NETCDF). Individual installation of these libraries is not practical, so the `HPC-Stack <https://github.com/NOAA-EMC/hpc-stack>`__ was developed as a central installation system to ensure that the infrastructure environment across multiple platforms is as similar as possible. Installation of the HPC-Stack is required to run the SRW App.
+
+Instructions
+-------------------------
+Users working on systems that fall under `Support Levels 2-4 <https://github.com/ufs-community/ufs-srweather-app/wiki/Supported-Platforms-and-Compilers>`__ will need to install the HPC-Stack the first time they try to build applications (such as the SRW App) that depend on it. Users can either build the HPC-Stack on their local system or use the centrally maintained stacks on each HPC platform if they are working on a Level 1 system. Before installing the HPC-Stack, users on both Linux and MacOS systems should set the stack size to "unlimited" (if allowed) or to the largest possible value:
+
+.. code-block:: console
+
+   # Linux, if allowed
+   ulimit -s unlimited
+
+   # MacOS, this corresponds to 65MB
+   ulimit -S -s unlimited
+
```

Comment:
@natalie-perlin Do you know? I think this is a holdover from the v1.0.1 docs, but I want to be sure before deleting.

Reply:
@gspetro @danielabdi-noaa - The current SRW code only prints diagnostics of the  current limits for Linux and MacOS:

`./ush/machine/linux.yaml:9:  PRE_TASK_CMDS: '{ ulimit -a; }'`
`./ush/machine/macos.yaml:9:  PRE_TASK_CMDS: '{ ulimit -a; }'`

Format shown above as (<filename>:<line_no>:  LINE)

It does indeed sets "ulimit -s unlimited" on Level 1 platforms, e.g.:
`./ush/machine/gaea.yaml:19:  PRE_TASK_CMDS: '{ ulimit -s unlimited; ulimit -a; }'`
 
We may hard-code it in the yaml machine files for Linux and Mac, or leave it as it is and as it is mentioned in the documentation. The "pro" for leaving as it is:  more transparency for the user about setting all to ulimited. The "pro" for hard-coding it in machine yaml files: it simplifies the documentation.

Reply:
The modulefile for the MacOS  has also has the following command which indeed does the same as suggested in the docs (and maybe the reason it was placed into the module).

execute{cmd="ulimit -S -s unlimited", modeA={"load"}}
 
 **NB:** the modulefiles and the default machine files, however, set the settings to the unlimited only before or during the SRW build and run. 
 The documentation mentions these settings to be done prior to the _HPC-stack installation_. So I think it needs to remain in the documentation.

Reply:
@natalie-perlin Do the HPC-Stack docs themselves say to do this? Maybe we can remove it because anything specific to HPC-Stack should be in their documentation... though if it's not there, adding it might be a challenge since there's not much support for the repo anymore.

Reply:
Based on my scan of the HPC-Stack docs, these instructions are not there. While I think this information would be better provided there, HPC-Stack is not really being maintained anymore, and getting a PR through could be a challenge, since spack-stack is the new priority. If it's okay with @MichaelLueken , I think we can just leave this for the minor release and remove it at the next release when we move to spack-stack.

Reply:
@gspetro-NOAA For the minor release, we should go ahead and keep this in.  Once the SRW App moves to spack-stack for the next release, this can be readdressed.